pkg/api/v20240812preview/openshiftcluster_validatestatic_test.go

```diff
@@ -1161,3 +1161,152 @@ func TestOpenShiftClusterStaticValidateDelta(t *testing.T) {
 
 	runTests(t, testModeUpdate, tests)
 }
+
+func TestOpenShiftClusterStaticValidatePlatformWorkloadIdentityProfile(t *testing.T) {
+	createTests := []*validateTest{
+		{
+			name: "valid empty workloadIdentityProfile",
+			modify: func(oc *OpenShiftCluster) {
+				oc.Properties.PlatformWorkloadIdentityProfile = nil
+			},
+		},
+		{
+			name: "valid workloadIdentityProfile",
+			modify: func(oc *OpenShiftCluster) {
+				oc.Properties.PlatformWorkloadIdentityProfile = &PlatformWorkloadIdentityProfile{
+					PlatformWorkloadIdentities: []PlatformWorkloadIdentity{
+						{
+							OperatorName: "FAKE-OPERATOR",
+							ResourceID:   "/subscriptions/12345678-1234-1234-1234-123456789012/resourceGroups/a-fake-group/providers/Microsoft.RedHatOpenShift/userAssignedIdentities/fake-cluster-name",
```

Comment:
should this be `Microsoft.ManagedIdentity/userAssignedIdentities`?
same for other test cases.

Reply:
The regexp the resourceID is being compared against is at [pkg/api/v20240812preview/openshiftcluster_validatestatic.go line 430](https://github.com/Azure/ARO-RP/pull/3563/files/ca9536ab55648929c0b9e424f31978bcbd1baada#diff-5a28605fb092ea2ef41d370f17daac8ce3c72e7c060a83d62f8aa0366bdfdd4aR430)

The specific provider for the resource ID isn't being checked, just that the string is formatted like a resource ID at all. I can change the value in the test cases to `Microsoft.ManagedIdentity` if you'd like